### PR TITLE
[auto parallel] Auto recompute enable

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR types
Others

### PR changes
Others

### Description
PCard-73145
auto recompute

### CODE REVIEW

**Review by zhaoyinglia — DISMISSED** [review]
LGTM

File: test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py [hunk 1]
```diff
@@ -780,16 +781,29 @@ def forward(
...
-            position_ids=position_ids,
-            attention_mask=attention_mask,
-            inputs_embeds=inputs_embeds,
-            use_cache=use_cache,
-            past_key_values=past_key_values,
-            output_attentions=output_attentions,
-            output_hidden_states=output_hidden_states,
-        )
+        if self.config.recompute:
+            outputs = recompute(
```
Comment (zhiqiu) [inline test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py]: why recompute whole llama

Comment (FeixLiu) [reply test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py]: done update

**Review by zhaoyinglia — DISMISSED** [review]
LTGM for completion and reshard

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**